pytensor/tensor/rewriting/linalg.py

```diff
@@ -433,17 +433,19 @@ def rewrite_det_diag_from_eye_mul(fgraph, node):
     if len(non_eye_inputs) >= 2:
         return None
 
-    # Checking if original x was matrix
-    if not non_eye_inputs[0].owner:
-        # It was a matrix
-        det_val = non_eye_inputs[0].diagonal().prod()
+    # Checking if original x was scalar/vector/matrix
+
+    if non_eye_inputs[0].type.broadcastable[-2:] == (True, True):
+        # For scalar
+        det_val = non_eye_inputs[0].owner.inputs[0] ** (eye_input[0].shape[0]).astype(
```

Comment:
Why are you trying to get the input (which may not even exist if non_eye_inputs[0] is a constant)?
```suggestion
        det_val = non_eye_inputs[0] ** (eye_input[0].shape[0]).astype(
```

Probably the same for the other branches

Reply:
In general you could have a scalar coming out of an arbitrary Op, and you don't know if the first input is the scalar value, imagine it comes out of `pt.random.normal(size=(1,1))`, the first input of that `Op` is a `rng` which doesn't even make sense. You can use the output, and if you want to get rid of the dummy dimensions, you can call `squeeze(axis=(-2, -1))`, assuming you even need to do that

Reply:
I still don't get why you're doing `some_input.owner.inputs[0]`?

I suspect you are assuming it comes from some dimshuffle that expands dims, but you haven't checked it and it could come out of many other operations where just using the first input doesn't make sense.

Also you don't need to, you can call squeeze to safely make a (1, 1) matrix back to a scalar and keep the rest of your code the same.

If that squeeze on top of the non eye input van be simplified further (like squeeze/expand) other rewrites will take care of it.